sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -903,15 +926,65 @@ private[joins] object LongHashedRelation {
       if (!rowKey.isNullAt(0)) {
         val key = rowKey.getLong(0)
         map.append(key, unsafeRow)
+      } else if (isNullAware) {
+        return new EmptyHashedRelationWithAllNullKeys
       }
     }
     map.optimize()
     new LongHashedRelation(numFields, map)
   }
 }
 
+/**
+ * A special HashedRelation indicates it built from a empty input:Iterator[InternalRow].
+ */
+class EmptyHashedRelation extends HashedRelation with Externalizable {
```

Comment:
shall we have a common trait for these 2 to contain the fake implementation?

Reply:
Ok.